src/orquestra/opt/optimizers/pso/continuous_pso_optimizer.py

```diff
@@ -0,0 +1,333 @@
+        particles in the swarm.
+
+        Parameters
+        ----------
+        positions : np.ndarray
+            (N, D) array of the current positions of the particles in the swarm.
+        velocities : np.ndarray
+            (N, D) array of the current velocities of the particles in the swarm.
+        best_positions : np.ndarray
+            (N, D) array of the best positions the particles in the swarm that
+            have been seen.
+        topology : SwarmTopology
+            Topology object that dictates the best swarm position.
+        """
+        dimensions = positions.shape[-1]
+        inertia_term = self.inertia * velocities
+        # The affinity terms are quite random, as they change for every dimension,
+        # every particle and every call to this function. Maybe we could have levels
+        # of randomness here.
+        affinity_towards_best_particle_position_term = (
+            self.random_number_generator.uniform(
+                0,
+                self.affinity_towards_best_particle_position,
+                (self.swarm_size, dimensions),
+            )
+            * np.subtract(best_positions, positions)
+        )
+        affinity_towards_best_swarm_position_term = (
+            self.random_number_generator.uniform(
+                0,
+                self.affinity_towards_best_swarm_position,
+                (self.swarm_size, dimensions),
+            )
+            * np.subtract(topology.best_swarm_position, positions)
+        )
+        velocities[:] = (
+            inertia_term
+            + affinity_towards_best_particle_position_term
+            + affinity_towards_best_swarm_position_term
+        )
+
+        if self.velocity_bounds is not None:
+            velocities[:] = np.clip(
+                velocities, self.velocity_bounds[0], self.velocity_bounds[1]
+            )
+        positions[:] = positions + self.learning_rate * velocities
+
+    def _minimize(
+        self,
+        cost_function: CostFunction,
+        initial_params: np.ndarray,
+        keep_history: bool = False,
+    ) -> OptimizeResult:
+        dimensions = initial_params.shape[-1]
+        if initial_params.ndim == 2:
+            assert initial_params.shape[0] == self.swarm_size, (
+                "If you provide an (N, D) array as initial parameters for the swarm, "
+                "N must equal the size of the swarm."
+            )
+            self.positions = initial_params
+        else:
+            self.positions = self.get_initial_params(dimensions)
+            # Set first particle in the swarm to the initial parameters:
+            self.positions[0, :] = initial_params
+        self.best_positions = self.positions.copy()
+        self.topology = self.topology_constructor(dimensions)
+        self.velocities = self.get_initial_velocities(dimensions)
+
+        n_iterations_since_last_improvement = 0
+        best_swarm_value_checkpoint = np.infty
```

Comment:
Please change to `np.inf`, see previous comments.

Reply:
Updated